### PR TITLE
Relocate Cookie

### PR DESCRIPTION
**Pull request by mostafakhudair**
Moving cookie class to main system folder, while its a single component and doesn't depends on http class

- [x] Component(s) with PHPdocs
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Cookie/CloneableInterface.php [hunk 1]
```diff
@@ -9,15 +9,15 @@
...
+namespace CodeIgniter\Cookie;
 
 use DateTimeInterface;
 
 /**
  * Interface for a fresh Cookie instance with selected attribute(s)
  * only changed from the original instance.
  */
-interface CloneableCookieInterface extends CookieInterface
+interface CloneableInterface extends CookieInterface
```
Comment (paulbalandan) [inline system/Cookie/CloneableInterface.php:20]: Please revert the name. `CloneableInterface` feels incomplete. Cloneable what? The current name is desired.

File: system/Cookie/Cookie.php [hunk 2]
```diff
@@ -36,7 +36,7 @@
  * $cookie2->getName(); // prod_cookie
  * ```
  */
-class Cookie implements ArrayAccess, CloneableCookieInterface
+class Cookie implements ArrayAccess, CloneableInterface
```
Comment (paulbalandan) [inline system/Cookie/Cookie.php:39]: See comment above

**Review by paulbalandan — COMMENTED** [review]
My contention for grouping the Cookie under the HTTP is simply because it is an HTTP concept. It is the representation of an _HTTP_ cookie. But if others would like to see the cookie in a separate folder, I'll gladly oblige.

Other comment is on renaming of `CloneableCookieInterface`. See below.

**Review by MGatner — COMMENTED** [review]
I like this! HTTP layer is bloated, and cleaning it up as much as possible is a good thing. I agree with the interface name but otherwise his looks great.

**Review by MGatner — CHANGES_REQUESTED** [review]
In addition to the changes already noted please recreate the current classes as deprecated extensions of the new ones.

**Comment by paulbalandan** [comment]
> I like this! HTTP layer is bloated, and cleaning it up as much as possible is a good thing. I agree with the interface name but otherwise his looks great.
> 
> > simply because it is an HTTP concept
> 
> @paulbalandan I understand the interest but I think we have precedence in this with a number of other components (like Honeypot, Security, Throttle...). With the goal of moving towards PSR-7 compliance I think having HTTP stick to PSR components would be a logical separation.

Hmm. Ok then. We'll have it as a separate component. Aside from that, there's no need to deprecate as this is in `>4.1.1`. Just change the changelog entry. I think the namespace there is written as `HTTP\Cookie`

File: system/Cookie/CloneableInterface.php (hunk 1 above)
Comment (mostafakhudair) [reply system/Cookie/CloneableInterface.php:20]: Cloneable just like Countable Interface, But i agree it looks incomplete

**Comment by mostafakhudair** [comment]
@MGatner CloneableInterface or CloneableCookieInterface?

**Comment by MGatner** [comment]
I vote for `CloneableCookieInterface` as it is clearer what the interface is for.

**Review by paulbalandan — APPROVED** [review]

**Pull request merged**